### PR TITLE
[Introspection] Allow introspection to test CAMetalLayer when xcode 11 and simulator.

### PR DESCRIPTION
**Pull request by mandel-macaque**
The order of the check matters, we only skip with a sim && not on
xcode11.

The other whay around will do the wrong short circuit. Since if we are
not on Xcode11 we will always skip, including device, which is not
correct.

Fixes: https://github.com/xamarin/xamarin-macios/issues/6243.

### CODE REVIEW

**Comment by mandel-macaque** [comment]
Please do test this in an older version of the OS, I do not have any old machine (at the office at least).

File: tests/introspection/iOS/iOSApiCtorInitTest.cs [hunk 1]
```diff
@@ -157,7 +157,7 @@ protected override bool Skip (Type type)
 
 			// Metal is not available on the (iOS8) simulator
 			case "CAMetalLayer":
-				return (Runtime.Arch == Arch.SIMULATOR);
+				return  (Runtime.Arch == Arch.SIMULATOR) && !TestRuntime.CheckXcodeVersion (11, 0);
```
Comment (wachaudh) [inline tests/introspection/iOS/iOSApiCtorInitTest.cs]:
```suggestion
				return (Runtime.Arch == Arch.SIMULATOR) && !TestRuntime.CheckXcodeVersion (11, 0);
```

**Comment by mandel-macaque** [comment]
Screenshot 2020-01-05 at 18 25 39
Above are the results on Mojave 10.14.5

**Comment by mandel-macaque** [comment]
Known issues: https://github.com/xamarin/maccore/issues/581

**Pull request merged**